### PR TITLE
chore(swift): use Hashable struct

### PR DESCRIPTION
**Pull request by Fluf22**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-1633]

### Changes included:

- remove tests that use a trick to test specific types not applicable to Swift
- update playground
- fix an issue with the insights spec
- add tests for the complex types with discriminators

[DI-1633]: https://algolia.atlassian.net/browse/DI-1633?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: templates/swift/modelObject.mustache [hunk 1]
```diff
@@ -1,4 +1,4 @@
...
+{{^objcCompatible}}{{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}public{{/nonPublicApi}} {{#useClasses}}final class{{/useClasses}}{{^useClasses}}struct{{/useClasses}} {{{classname}}}{{#vendorExtensions.x-has-child-generic}}<T: Codable{{#vendorExtensions.x-swift-hashable}} & Hashable{{/vendorExtensions.x-swift-hashable}}>{{/vendorExtensions.x-has-child-generic}}{{^vendorExtensions.x-has-child-generic}}{{#vendorExtensions.x-propagated-generic}}<T: Codable>{{/vendorExtensions.x-propagated-generic}}{{/vendorExtensions.x-has-child-generic}}: Codable{{#useJsonEncodable}}, JSONEncodable{{/useJsonEncodable}}{{#vendorExtensions.x-swift-hashable}}, Hashable{{/vendorExtensions.x-swift-hashable}} {
```
Comment (millotp) [inline templates/swift/modelObject.mustache:1]: I thought you wouldn't require Hashable for generic types, as you want

**Review by millotp — DISMISSED** [review]
looks good !

File: templates/swift/modelObject.mustache (hunk 1 above)
Comment (millotp) [reply templates/swift/modelObject.mustache:1]: why not ?

Comment (Fluf22) [reply templates/swift/modelObject.mustache:1]: That's exactly the discussion we had two days ago 😂😂
No it's not mandatory, yes it's a constraint
It just allows to have Hashable (and consequently Equatable) types 😁

Comment (millotp) [reply templates/swift/modelObject.mustache:1]: I think there was a misunderstandings during the discussion, if you add `Hashable` here, it forces the user to make their custom types Hashable right ? or is swift this weird ?

Comment (Fluf22) [reply templates/swift/modelObject.mustache:1]: It forces the user to make their custom types Hashable, indeed
I can remove this altogether, though!

Comment (Fluf22) [reply templates/swift/modelObject.mustache:1]: Found a solution, gonna push a fix

**Review by millotp — APPROVED** [review]
huge gg !

**Pull request merged**